### PR TITLE
Adds ShadowCasterVisitor as easy override by user

### PR DESCRIPTION
**Pull request by Kuranes**
Allows either override with user provided visitor, false (disable any visitor) or defaut visitor (which handles lights, subscene with rtt, transparent nodes)

### CODE REVIEW

File: sources/osgShadow/ShadowMap.js [hunk 1]
```diff
@@ -236,7 +152,14 @@ define( [
 
         // Overridable Visitor so that user can override the visitor to enable disable
         // in its own shadowmap implementation
-        this._removeNodesNeverCastingVisitor = new RemoveNodesNeverCastingVisitor( this._castsShadowTraversalMask );
+        // settings.userShadowCasterVisitor:
+        // - undefined means using default
+        // - false means no removal visitor needed
+        // - otherwiser must be an instance of a class inherited from shadowCaster
+        if ( settings.userShadowCasterVisitor !== false ) {
```
Comment (Kuranes) [inline sources/osgShadow/ShadowMap.js]: As discussed before, settings is considerer a dictionnary not an object.
But can add an accessor if feeling is strong on that one.

**Comment by cedricpinson** [comment]
do we need this feature now ?

**Comment by Kuranes** [comment]
It fixes a bug in transparent shadow casting ?

**Comment by cedricpinson** [comment]
I was talking about the visitor stuff, but ok never mind

**Comment by Kuranes** [comment]
On osgjs side we still need the visitor to remove transparent material otherwise default rendering on par with osgjs material won't give good results.

 ( it doesn't add new code, the visitor is the old one we had before. it's just allowing user control/overwrite of that part )

**Pull request merged**